### ISSUE TITLE
conditional implementation

### ISSUE THREAD

**Issue by PieraRiccio** [lucidrains/stylegan2-pytorch#194]
Hi, did you also implement a conditional implementation, so that I can pass files with labels, as in the original TensorFlow implementation?

**Comment by lucidrains** [comment]
@PieraRiccio Hey Piera! I haven't implemented the conditional part yet

I would have to think about how this could work as a CLI before moving forward. Perhaps I could ask for the labels as some standard data format that matches the filename of the images, and then have an extra flag for passing that in the labels file

**Comment by PieraRiccio** [comment]
@lucidrains Hi. :) Thank you for your answer. I think that a usual modality to do that is to retrieve the labels from subfolders of the main images folder. What do you think about this?

In order to run conditional training in the official TF implementation, I used the code shared in this issue, maybe it can be helpful:
https://github.com/NVlabs/stylegan2-ada/issues/22

**Comment by lucidrains** [comment]
Sure, that works! How do you have your files -> label organized at the moment?

**Comment by PieraRiccio** [comment]
I have a main folder and 4 sub-folders (which correspond to the 4 labels) :)

**Comment by lucidrains** [comment]
@PieraRiccio oh that may work! let me think about it before executing the code

**Comment by lucidrains** [comment]
@PieraRiccio so I looked over the official code, and it seems like labels are consumed by the generator and discriminator in the following way

in both G and D, the label id is first projected to the feature map size and normalized

in the generator, it is concatenated to the latent vector before being passed through the mapping network

in the discriminator, it seems to be also passed through a separate mapping network https://github.com/NVlabs/stylegan2-ada/blob/main/training/networks.py#L520, but the number of layers is defaulted to 0, so I'm not sure if it is being used. in the end, it is multiplied with the image embedding just before the output https://github.com/NVlabs/stylegan2-ada/blob/main/training/networks.py#L619

**Comment by foobar8675** [comment]
@lucidrains cc @PieraRiccio I have a very dirty attempt at the conditional logic https://github.com/foobar8675/stylegan2-pytorch/tree/dev, running on cifar100 right now.

**Comment by lucidrains** [comment]
@foobar8675 Amazing Matthew!

**Comment by PieraRiccio** [comment]
@foobar8675 Thank you! :)

**Comment by foobar8675** [comment]
I've been trying to run on cifar100, which was probably not realistic (only 600 images / class) . starting again with cifar10

**Comment by PieraRiccio** [comment]
@foobar8675 Given the small number of images for each label, have you also tried using adaptive discriminator augmentation? :)

**Comment by foobar8675** [comment]
@PieraRiccio thanks for the suggestion. I have not, but I do have augmentation set to .25.  My understanding, which might be wrong, is that ADA will kick in with augmentation if it senses augmentation is needed and otherwise not augment. since I have augmentation always set, I'm not sure if it would help? 

However, I might be misunderstanding. Do you have any thoughts?

**Comment by BrentOGara** [comment]
@matthewchung74 This: https://github.com/lucidrains/stylegan2-pytorch/issues/194#issuecomment-753704441 looks very cool, exactly what I need... but the link no longer works. Is this still available anywhere? Thanks.

**Comment by matthewchung74** [comment]
@BrentOGara unfortunately, i deleted that repo, since it wasn't quite as good as what I wanted. if you are looking for a conditional stylegan2, nvidia's implementation supports it. https://github.com/NVlabs/stylegan2-ada-pytorch

**Comment by BrentOGara** [comment]
@matthewchung74 Thank you for the link, I'll see if I can figure it out. I'm still very new to this field.
